drivers/timer/nrf_rtc_timer.c

```diff
@@ -660,6 +660,23 @@ uint32_t sys_clock_cycle_get_32(void)
 	return (uint32_t)z_nrf_rtc_timer_read();
 }
 
+void sys_clock_disable(void)
+{
+	uint32_t mask = NRF_RTC_INT_TICK_MASK     |
+			NRF_RTC_INT_OVERFLOW_MASK |
+			NRF_RTC_INT_COMPARE0_MASK |
+			NRF_RTC_INT_COMPARE1_MASK |
+			NRF_RTC_INT_COMPARE2_MASK |
+			NRF_RTC_INT_COMPARE3_MASK;
+
+	nrf_rtc_task_trigger(RTC, NRF_RTC_TASK_STOP);
+	nrf_rtc_task_trigger(RTC, NRF_RTC_TASK_CLEAR);
+	irq_disable(RTC_IRQn);
+	nrf_rtc_int_disable(RTC, mask);
+	nrf_rtc_event_disable(RTC, mask);
```

Comment:
This could be extracted to a helper function called by both `sys_clock_disable()` and `sys_clock_driver_init()`.

Reply:
Moved to separate function